lib/common/mainloop.c

```diff
@@ -1280,10 +1280,7 @@ mainloop_child_add_with_flags(pid_t pid, int timeout, const char *desc, void *pr
     child->privatedata = privatedata;
     child->callback = callback;
     child->flags = flags;
-
-    if(desc) {
-        child->desc = strdup(desc);
-    }
+    pcmk__copy_or_null(&child->desc, desc);
```

Comment:
no value in thinking over if g_new does initialize memory as this is anyway going away with the next commit.